### PR TITLE
194 misleading errors fix

### LINKED ISSUE #194: misleading error message
Need to fix error messages in the contracts

### LINKED ISSUE #199: Wrong error message when delegating twice in the same epoch `DelegationUtils`.

### PR DESCRIPTION
**Pull request by BritikovKI**
Closes #194 
Closes #199

### CODE REVIEW

File: packages/api3-voting/test/delegation-integration.js [hunk 1]
```diff
@@ -95,14 +95,14 @@ contract('API3 Voting App delegation tests', ([root, voter1, voter2, voter3, non
...
-                "Unauthorized"
+                "This address un/delegated less then a week before"
             );
         });
 
         it('undo delegate earlier then after a week', async () => {
             await expectRevert(
                 pool.undelegateVotingPower({from: voter1}),
-                "Unauthorized"
+                "This address un/delegated less then a week before"
```
Comment (kinspotting) [inline packages/api3-voting/test/delegation-integration.js:105]: pls change to `less than`

File: packages/pool/contracts/GetterUtils.sol [hunk 2]
```diff
@@ -336,13 +339,13 @@ abstract contract GetterUtils is StateUtils, IGetterUtils {
         }
         // Revert if the value being searched for comes before
         // `minimumCheckpointIndex`
-        require(i == 0, ERROR_VALUE);
+        require(i == 0, NOT_FOUND_CHECKPOINT);
```
Comment (kinspotting) [inline packages/pool/contracts/GetterUtils.sol:342]: pls change to `CHECKPOINT_NOT_FOUND`

File: packages/pool/contracts/GetterUtils.sol [hunk 3]
```diff
@@ -6,6 +6,9 @@ import "./interfaces/IGetterUtils.sol";
 
 /// @title Contract that implements getters
 abstract contract GetterUtils is StateUtils, IGetterUtils {
+
+    string private constant NOT_FOUND_CHECKPOINT = "API3DAO.GetterUtils: VALUE CANNOT BE FOUND AFTER PROVIDED CHECKPOINT";
```
[the remark this answers is not in the document]
Comment (kinspotting) [reply packages/pool/contracts/GetterUtils.sol:10]: pls change to `CHECKPOINT_NOT_FOUND`

File: packages/pool/contracts/GetterUtils.sol [hunk 4]
```diff
@@ -376,7 +379,7 @@ abstract contract GetterUtils is StateUtils, IGetterUtils {
                 return 0;
             }
             else {
-                revert(ERROR_VALUE);
+                revert(NOT_FOUND_CHECKPOINT);
```
Comment (kinspotting) [inline packages/pool/contracts/GetterUtils.sol:382]: same

File: packages/pool/contracts/StakeUtils.sol [hunk 5]
```diff
@@ -6,6 +6,12 @@ import "./interfaces/IStakeUtils.sol";
 
 /// @title Contract that implements staking functionality
 abstract contract StakeUtils is TransferUtils, IStakeUtils {
+
+
+    string private constant NOT_ENOUGH_FUNDS = "API3DAO.StakeUtils: USER DONT HAVE ENOUGH TOKENS TO STAKE/UNSTAKE THE PROVIDED AMOUNT";
+    string private constant UNSTAKE_TIMING_ERROR = "API3DAO.StakeUtils: USER IS ALLOWED TO UNSTAKE NO LONGER THEN A WEEK FROM SCHEDULED TIME";
+    string private constant STAKING_ADDRESS_ERROR = "API3DAO.StakeUtils: IT IS ONLY POSSIBLE TO STAKE TO YOURSELF";
```
Comment (kinspotting) [inline packages/pool/contracts/StakeUtils.sol:13]: pls move `ERROR` to the front for the consistency, so we can have something like `ERROR_UNSTAKE_TIMING`, `ERROR_NOT_ENOUGH_FUNDS`

Comment (kinspotting) [reply packages/pool/contracts/StakeUtils.sol:13]: etc.

File: packages/pool/contracts/TimelockUtils.sol [hunk 6]
```diff
@@ -8,6 +8,13 @@ import "./interfaces/ITimelockUtils.sol";
 /// @dev TimelockManager contracts interface with this contract to transfer
 /// API3 tokens that are locked under a vesting schedule.
 abstract contract TimelockUtils is ClaimUtils, ITimelockUtils {
+
+    string private constant INVALID_TIME_OR_AMOUNT =
+    "API3DAO.TimelockUtils: AMOUNT SHOULD BE GREATER THEN 0 AND releaseEnd > releaseStart";
+    string private constant LOCKED_TOKENS_ERROR = "API3DAO.TimelockUtils: USER SHOULDN'T HAVE TIMELOCKED TOKENS";
```
Comment (kinspotting) [inline packages/pool/contracts/TimelockUtils.sol:14]: pls move `ERROR` to the front

File: packages/pool/contracts/TransferUtils.sol [hunk 7]
```diff
@@ -6,6 +6,12 @@ import "./interfaces/ITransferUtils.sol";
 
 /// @title Contract that implements token transfer functionality
 abstract contract TransferUtils is DelegationUtils, ITransferUtils {
+
+    string private constant WRONG_TOTAL_FUNDS =
+    "API3DAO.TransferUtils: USER TOTAL FUNDS SHOULD BE BIGGER THEN LOCKED AND AMOUNT TO WITHDRAW";
+    string private constant TOO_BIG_AMOUNT =
```
Comment (kinspotting) [inline packages/pool/contracts/TransferUtils.sol:12]: `AMOUNT_TOO_BIG`

File: packages/pool/contracts/TransferUtils.sol [hunk 8]
```diff
@@ -6,6 +6,12 @@ import "./interfaces/ITransferUtils.sol";
...
+    "API3DAO.TransferUtils: WITHDRAWAL AMOUNT SHOULD BE LESS OR EQUAL TO THE UNSTAKED TOKENS";
```
Comment (kinspotting) [inline packages/pool/contracts/TransferUtils.sol:13]: same

File: packages/pool/contracts/DelegationUtils.sol [hunk 9]
```diff
@@ -7,7 +7,13 @@ import "./interfaces/IDelegationUtils.sol";
...
 abstract contract DelegationUtils is RewardUtils, IDelegationUtils {
 
-    string internal constant ERROR_DELEGATION_BALANCE = "Cannot delegate zero shares";
+    string internal constant ERROR_DELEGATION_BALANCE = "API3DAO.DelegationUtils: Cannot delegate zero shares";
+    string internal constant ERROR_DELEGATION_ADRESSES =
+    "API3DAO.DelegationUtils: Cannot delegate to zero addresses, to yourself and if you've already delegated";
+    string internal constant ERROR_DELEGATED_RECENTLY =
+    "API3DAO.DelegationUtils: This address un/delegated less then a week before";
+    string internal constant ERROR_ACTIVE_RECENTLY =
+    "API3DAO.DelegationUtils: This address voted or made a proposal less then a week before";
```
Comment (kinspotting) [inline packages/pool/contracts/DelegationUtils.sol:16]: pls cahnge to `less than`

File: packages/pool/contracts/DelegationUtils.sol [hunk 10]
```diff
@@ -7,7 +7,13 @@ import "./interfaces/IDelegationUtils.sol";
 /// @title Contract that implements voting power delegation
 abstract contract DelegationUtils is RewardUtils, IDelegationUtils {
 
-    string internal constant ERROR_DELEGATION_BALANCE = "Cannot delegate zero shares";
+    string internal constant ERROR_DELEGATION_BALANCE = "API3DAO.DelegationUtils: Cannot delegate zero shares";
+    string internal constant ERROR_DELEGATION_ADRESSES =
+    "API3DAO.DelegationUtils: Cannot delegate to zero addresses, to yourself and if you've already delegated";
```
Comment (kinspotting) [inline packages/pool/contracts/DelegationUtils.sol:12]: pls change to `Cannot delegate to yourself or zero address`

**Review by kinspotting — CHANGES_REQUESTED** [review]
needs changes

File: packages/api3-voting/test/delegation-integration.js (hunk 1 above)
Comment (BritikovKI) [reply packages/api3-voting/test/delegation-integration.js]: Done

File: packages/pool/contracts/ClaimUtils.sol [hunk 11]
```diff
@@ -6,9 +6,14 @@ import "./interfaces/IClaimUtils.sol";
 
 /// @title Contract that implements the insurance claim payout functionality
 abstract contract ClaimUtils is StakeUtils, IClaimUtils {
+
+
+    string private constant ERROR_CLAIM_AMOUNT = "API3DAO.ClaimUtils: TOTAL STAKE SHOULD BE BIGGER THEN CLAIM AMOUNT";
```
[the remark this answers is not in the document]
Comment (BritikovKI) [reply packages/pool/contracts/ClaimUtils.sol]: Fixed

File: packages/pool/contracts/GetterUtils.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (BritikovKI) [reply packages/pool/contracts/GetterUtils.sol]: Fixed

File: packages/pool/contracts/GetterUtils.sol (hunk 2 above)
Comment (BritikovKI) [reply packages/pool/contracts/GetterUtils.sol]: Fixed

File: packages/pool/contracts/GetterUtils.sol (hunk 4 above)
Comment (BritikovKI) [reply packages/pool/contracts/GetterUtils.sol]: Fixed

File: packages/pool/contracts/StakeUtils.sol (hunk 5 above)
Comment (BritikovKI) [reply packages/pool/contracts/StakeUtils.sol]: Fixed

File: packages/pool/contracts/TimelockUtils.sol (hunk 6 above)
Comment (BritikovKI) [reply packages/pool/contracts/TimelockUtils.sol]: Fixed

File: packages/pool/contracts/TransferUtils.sol (hunk 7 above)
Comment (BritikovKI) [reply packages/pool/contracts/TransferUtils.sol]: Fixed

File: packages/pool/contracts/TransferUtils.sol [hunk 12]
```diff
@@ -6,6 +6,12 @@ import "./interfaces/ITransferUtils.sol";
 
 /// @title Contract that implements token transfer functionality
 abstract contract TransferUtils is DelegationUtils, ITransferUtils {
+
+    string private constant WRONG_TOTAL_FUNDS =
+    "API3DAO.TransferUtils: USER TOTAL FUNDS SHOULD BE BIGGER THEN LOCKED AND AMOUNT TO WITHDRAW";
```
[the remark this answers is not in the document]
Comment (BritikovKI) [reply packages/pool/contracts/TransferUtils.sol]: Fixed

File: packages/pool/contracts/TransferUtils.sol (hunk 8 above)
Comment (BritikovKI) [reply packages/pool/contracts/TransferUtils.sol]: Fixed

File: packages/pool/contracts/DelegationUtils.sol (hunk 9 above)
Comment (BritikovKI) [reply packages/pool/contracts/DelegationUtils.sol]: Fixed

**Review by kinspotting — APPROVED** [review]
lgtm

**Pull request merged**